cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowHttpHandler.java

```diff
@@ -129,30 +131,11 @@ public void resumeWorkflowRun(HttpRequest request, final HttpResponder responder
       }
       controller.resume().get();
       responder.sendString(HttpResponseStatus.OK, "Program run resumed.");
-    }  catch (SecurityException e) {
-      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
-    } catch (Throwable e) {
-      LOG.error("Got exception:", e);
-      responder.sendStatus(HttpResponseStatus.INTERNAL_SERVER_ERROR);
-    }
-  }
-
-  private void sendInvalidResponse(HttpResponder responder, Id.Program id) {
-    try {
-      AppFabricServiceStatus status;
-      ProgramStatus programStatus = getProgramStatus(id, ProgramType.WORKFLOW);
-      if (programStatus.getStatus().equals(HttpResponseStatus.NOT_FOUND.toString())) {
-        status = AppFabricServiceStatus.PROGRAM_NOT_FOUND;
-      } else if (ProgramController.State.COMPLETED.toString().equals(programStatus.getStatus())
-          || ProgramController.State.KILLED.toString().equals(programStatus.getStatus())
-          || ProgramController.State.ERROR.toString().equals(programStatus.getStatus())) {
-        status = AppFabricServiceStatus.PROGRAM_ALREADY_STOPPED;
-      } else {
-        status = AppFabricServiceStatus.RUNTIME_INFO_NOT_FOUND;
-      }
-      responder.sendString(status.getCode(), status.getMessage());
     } catch (SecurityException e) {
       responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
+    } catch (NotFoundException e) {
+      LOG.warn("NotFoundException while runtime information for run id is not found.", e);
+      responder.sendString(HttpResponseStatus.NOT_FOUND, e.getMessage());
     } catch (Throwable e) {
```

Comment:
You can remove these catch-blocks (SecurityException, NotFoundException, Throwable) since HttpExceptionHandler is handling them already.